pkg/gardenlet/controller/seed/seed/reconciler_delete.go

```diff
@@ -272,6 +273,12 @@ func (r *Reconciler) runDeleteSeedFlow(
 			Dependencies: flow.NewTaskIDs(destroyFluentOperatorResources, noControllerInstallations),
 			SkipIf:       seedIsGarden,
 		})
+		destroyGardenletVPA = g.Add(flow.Task{
```

Comment:
Why this additional complexity? Today, we destroy VPA components (if we manage them at all) in parallel to everything else (i.e., we don't have any dependency to other tasks). So, I wouldn't now add one for `destroyGardenletVPA` while not checking all the others.

Generally, we don't care whether all VPAs are deleted when deleting the VPA components, or is there a reason why we should? When VPA is managed externally, we cannot control this anyways.

Reply:
Ok, let's then keep it as it is. 

I was thinking for the case when VPA components are destroyed before deleting the VPA. This would leave an orphaned VerticalPodAutoscalerCheckpoint resource. But it is another problem that VPA resource does not have a finalizer or VerticalPodAutoscalerCheckpoint does not have an ownerReference to the owning VPA resource.

It was also a comment that with this ordering we rely that VPAs can be deleted without requiring the VPA components to be running. If VPA components were maintaining a finalizer on the VPA resource, then this parallel deletion of resource and controller wouldn't work.

Anyways, I agree that it is additional complexity, so let's keep it as it is..

Reply:
OK, thank you. I think adding an owner reference in `VerticalPodAutoscalerCheckpoint` makes sense. Adding a finalizer maybe not much, given that no external infrastructure is created.
This way, the deletion would always be independent of the VPA components (which is preferred, I guess0>

Reply:
Another upstream VPA improvement 😂